### DISCUSSION TITLE
MDX Integration

### DISCUSSION THREAD

**Discussion by GridexX** [marp-team/marp#177] [Ideas]
MDX is a new extension for Markdown that authorize writing `JSX` in markdown files. See doc [here](https://mdxjs.com/) !

It would be nice to have the possibility to also convert `MDX` files into slides :smile:

**Comment by yhatt** [comment]
MDX is definitely a great flavor, especially for React developers. On the other hand, we have taken a dim view from the point of Marp.

- Marp has designed for taking the simplest approaches to make traditional static slides, like PDF presentation. MDX is a bit much option for this purpose.
  - Marp team doesn't like getting powered by client-side JS (e.g. React) in slides, as described in [Marpit documentation "JavaScript not required"](https://marpit.marp.app/inline-svg?id=javascript-not-required).

- MDX may require some of a bit deep contexts about front-end development: A knowledge of view framework such as React, JavaScript build system for compiling MDX to JS, etc. This prerequisite is not match to our purpose "Everyone has only to write the story".

- MDX conversion is backed by [remark](https://github.com/remarkjs/remark) engine while Marp is based on [markdown-it](https://github.com/markdown-it/markdown-it) engine. This difference really makes getting MDX support difficult because markdown-it and its ecosystem have not covered MDX.
  - We had already suggested transition to remark engine for MDX support 2 years ago, but have decided not to take that because of less positive feedbacks: marp-team/marpit#194

MDX ecosystem already has solutions for creating slides that looks like popular than Marp (over 3k stars): [mdx-deck] (over 10k stars), [fusuma] (over 5k stars). You may have to check them first if you missed.

[mdx-deck]: https://github.com/jxnblk/mdx-deck
[fusuma]: https://github.com/hiroppy/fusuma

MDX support would not come as "Marp" toolset for these reasons, but "Marpit" framework is not giving up a possibility of MDX integration yet. This framework aims to become a scaffolding library for various slide frameworks, and we think should not lock in to a single way of Marp. [We're welcome feedback to RFC for MDX support!](https://github.com/marp-team/marpit/issues/194)
